samza-test/src/test/scala/org/apache/samza/test/integration/StreamTaskTestUtil.scala

```diff
@@ -122,12 +104,36 @@ object StreamTaskTestUtil {
     TestTask.reset()
   }
 
+  var servers: Buffer[KafkaServer] = null
+
   def beforeSetupServers {
-    zookeeper = new EmbeddedZookeeper(zkConnect)
-    server1 = TestUtils.createServer(new KafkaConfig(props1))
-    server2 = TestUtils.createServer(new KafkaConfig(props2))
-    server3 = TestUtils.createServer(new KafkaConfig(props3))
-    zkClient = new ZkClient(zkConnect + "/", 6000, 6000, ZKStringSerializer)
+    zookeeper = new EmbeddedZookeeper()
+    zkUtils = ZkUtils(zkConnect, zkSessionTimeout, zkConnectionTimeout, JaasUtils.isZkSecurityEnabled())
+
+    val props = TestUtils.createBrokerConfigs(3, zkConnect, true)
```

Comment:
Kafka distrib downloaded by zopkio is [kafka_2.10-0.10.0.1.tgz](http://archive.apache.org/dist/kafka/0.10.0.1/kafka_2.10-0.10.0.1.tgz) from Apache repositories.
The 0.10.x protocol for both internal and client communication [is enabled by default](https://kafka.apache.org/documentation.html#upgrade_10) for this version. Kafka lets you lower it for compatibility/upgrade purpose (our case) but the Kafka 10 client is fortunately backward compatible.